src/software/gui/drawing/field.cpp

```diff
@@ -57,14 +63,52 @@ void drawField(QGraphicsScene* scene, const Field& field)
     drawCenterCircle(scene, field, pen);
     drawDefenseAreas(scene, field, pen);
     drawGoals(scene, field, pen);
+}
+
+void drawTeamGoalText(QGraphicsScene* scene, const Field& field)
+{
+    QGraphicsSimpleTextItem* friendly_text = new QGraphicsSimpleTextItem("FRIENDLY");
+    QGraphicsSimpleTextItem* enemy_text    = new QGraphicsSimpleTextItem("ENEMY");
+    QFont sansFont("Helvetica [Cronyx]");
+    sansFont.setPointSizeF(1);
+    friendly_text->setFont(sansFont);
+    friendly_text->setBrush(Qt::black);
+    enemy_text->setFont(sansFont);
+    enemy_text->setBrush(Qt::black);
+
+    // Scale the text so it's width matches the goal area
+    double friendly_text_scaling_factor =
+        1.0 / (friendly_text->boundingRect().width() / field.goalYLength());
+    double enemy_text_scaling_factor =
+        1.0 / (enemy_text->boundingRect().width() / field.goalYLength());
```

Comment:
Wanted to check, in the video, the "ENEMY" text is much larger as is the opposing rectangle surrounding the goal. That's due to the defenders, right?

Reply:
The blue rectangle is the avoid area which is larger because we can't touch defenders partially in the defense area. It's unrelated to my changes.

The text is larger because it's scaled to match the width of the goal, but there are fewer characters so it scales "bigger"

Reply:
Kindly also asking for a follow-up on if you think having the ball's range lines be thicker would be beneficial.

Reply:
You mean the lines from the ball to the friendly goal? Personally I like them as they are since they're not too intrusive on other stuff that's more important.

Reply:
Was referring to the "ball to friendly goal" line, I'm fine with the current line thickness then.